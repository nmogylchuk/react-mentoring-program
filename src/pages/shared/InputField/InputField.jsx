import React from "react";
import PropTypes from "prop-types";
import 'pages/shared/InputField/InputField.scss';

function InputField({label, value, type, name, placeholder, handleMovieChange}) {
    return (
        <div className="input-field">
            <label className="input-field__label">{label}</label>
            <input
                className="input-field__input"
                value={value}
                type={type}
                name={name}
                placeholder={placeholder}
                onChange={handleMovieChange}
            />
        </div>
    )
}

InputField.propTypes = {
    label: PropTypes.string.isRequired,
    value: PropTypes.any.isRequired,
    type: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    placeholder: PropTypes.string.isRequired,
    handleMovieChange: PropTypes.func.isRequired,
};

export default InputField;
