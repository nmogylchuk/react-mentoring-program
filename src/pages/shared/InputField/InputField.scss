@import 'src/assets/styles/variables';

.input-field {
  margin-top: 5px;
  display: flex;
  flex-direction: column;

  &__label {
    color: $red;
    font-size: 1.2rem;
    text-transform: uppercase;
  }

  &__input {
    display: inline-block;
    min-height: 30px;
    border: none;
    color: $white;
    font-size: 1.6rem;
    background: $grey;
    padding: 10px 20px;
    margin: 10px 0;
    width: 100%;
  }
}
