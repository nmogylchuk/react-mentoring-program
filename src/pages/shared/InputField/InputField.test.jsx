import React from 'react';
import rerender from 'react-test-renderer';
import InputField from "./InputField";
import {jest} from "@jest/globals";

describe('when Input Field', () => {

    const movie = {
        "id": 447365,
        "title": "Guardians of the Galaxy Vol. 3",
        "tagline": "",
        "vote_average": 0,
        "vote_count": 9,
        "release_date": "2020-05-01",
        "poster_path": "https://image.tmdb.org/t/p/w500/ldoY4fTZkGISMidNw60GHoNdgP8.jpg",
        "overview": "The third film based on Marvel's Guardians of the Galaxy.",
        "budget": 0,
        "revenue": 0,
        "genres": [
            "Action",
            "Adventure",
            "Science Fiction"
        ],
        "runtime": null
    }

    test('then snapshot created', () => {
        const inputFieldComponent = rerender.create(<InputField label="Title" value={movie.title} type="text"
                                                                name="title"
                                                                placeholder="Title here"
                                                                handleMovieChange={jest.fn()}
                                                                onChange={jest.fn()}/>);
        const tree = inputFieldComponent.toJSON();
        expect(tree).toMatchSnapshot();
    });
});