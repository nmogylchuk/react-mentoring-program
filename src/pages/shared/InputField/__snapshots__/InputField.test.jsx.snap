// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when Input Field then snapshot created 1`] = `
<div
  className="input-field"
>
  <label
    className="input-field__label"
  >
    Title
  </label>
  <input
    className="input-field__input"
    name="title"
    onChange={[MockFunction]}
    placeholder="Title here"
    type="text"
    value="Guardians of the Galaxy Vol. 3"
  />
</div>
`;
